e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/edit-payment.spec.js

```diff
@@ -77,24 +77,54 @@ context(`${PDC_TEAMS.PDC_RECONCILE} users can edit payments`, () => {
     cy.visit(`/utilisation-reports/${reportId}`);
   });
 
-  it('should allow the user to navigate to the edit payment page', () => {
+  it('should allow the user to navigate to the edit payment page from the premium payments table', () => {
     pages.utilisationReportPage.premiumPaymentsTab.clickPaymentLink(paymentId);
 
-    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}`));
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}?redirectTab=premium-payments`));
+  });
+
+  it('should allow the user to navigate to the edit payment page from the payment details table', () => {
+    cy.visit(`/utilisation-reports/${reportId}#payment-details`);
+
+    pages.utilisationReportPage.paymentDetailsTab.clickPaymentLink(paymentId);
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}?redirectTab=payment-details`));
+  });
+
+  it('should return to the premium payments page when the back link is clicked after accessing the edit payment page from the premium payments table', () => {
+    pages.utilisationReportPage.premiumPaymentsTab.clickPaymentLink(paymentId);
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}?redirectTab=premium-payments`));
+
+    pages.utilisationReportEditPaymentPage.clickBackLink();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}#premium-payments`));
+  });
+
+  it('should return to the payment details page when the back link is clicked after accessing the edit payment page from the payment details table', () => {
+    cy.visit(`/utilisation-reports/${reportId}#payment-details`);
+
+    pages.utilisationReportPage.paymentDetailsTab.clickPaymentLink(paymentId);
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}/edit-payment/${paymentId}?redirectTab=payment-details`));
+
+    pages.utilisationReportEditPaymentPage.clickBackLink();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${reportId}#payment-details`));
```

Comment:
Rather than having a long single "it" block, should we split up like this?

```js
describe('when ...', () => {
  it('should ...', () => {
    
  });
});

If the current approach (single its) is used throughout then happy to keep as is so that it's aligned 👍 
```

Reply:
Refactored to use the describe blocks thanks